### DISCUSSION TITLE
Why not install or add the path to the user or system path!

### DISCUSSION THREAD

**Discussion by rmerriam** [xpack-dev-tools/gcc-xpack#4] [Q&A]
Why do you recommend not setting the path? Or providing an install to system directories?

**Comment by ilg-ul** [comment]
Mainly because you increase the risk to get unreproducible builds.

**Reply by rmerriam** [reply]
Why?

**Reply by ilg-ul** [reply]
Because on different machines you may have different versions installed in the global/system folders, or even on the same machine at different moments, and the same build may behave differently.

On the contrary, if you install a given version of the tool as a local dependency of a project, you can build the project on any machine and will always use the same version of the tool.

But if you have a good argument for such a use case, I'm open to consider it.

**Reply by rmerriam** [reply]
My specific situation is installing on a Raspberry Pi 4b where I need GCC 11.x for the C++20 support. I cross compile from my Linux Desktop. The supported GCC 10.x on the Pi doesn't have the versions of some libs (libc, libstdc++) required by the cross compilation. Easiest way to me is to install these builds as the system.

I just tried using 'alternatives' but is doesn't fix the library problem. I can see libstdc++.so.6 but not libc. Not sure you can use alternatives for libs.

**Reply by ilg-ul** [reply]
Ah, ok, I did not think that my binary xPacks will be used instead of the system tools.

Although not an intended use case, it still makes sense.

I'll try to update xpm to add support to install tools in the system, but it'll take a while.

For now you don't even have to use xpm, you can simply download the archive in a folder of your choice and make some links from a folder in your path.
